src/pages/workspaces/workspace/WorkspaceFiles.ts

```diff
@@ -0,0 +1,45 @@
+import _ from 'lodash/fp'
+import { div, h } from 'react-hyperscript-helpers'
+import * as breadcrumbs from 'src/components/breadcrumbs'
+import FileBrowser from 'src/components/file-browser/FileBrowser'
+import GCSFileBrowserProvider from 'src/libs/ajax/file-browser-providers/GCSFileBrowserProvider'
+import { forwardRefWithName } from 'src/libs/react-utils'
+import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer'
+import { useMemo } from 'use-memo-one'
+
+
+export const WorkspaceFiles = _.flow(
+  forwardRefWithName('WorkspaceFiles'),
```

Comment:
I know we use `forwardRefWithName` all over the place, but I can't figure out what it actually does (and there is no documentation). Do you happen to know?

Reply:
Since we use anonymous arrow functions, React can't figure out a component name for something like:
```js
const Component = forwardRef((props, ref) => ...)
```

`forwardRefWithName` adds a name to the component function so that it is identifiable in the Components panel of the React dev tools.

These would have the same effect:
```js
const Component = forwardRefWithName('MyComponent', (props, ref) => ...)
```

```js
const Component = forwardRef(function MyComponent(props, ref) { ... })
```

Reply:
As for why it's needed here (despite `ref` being unused), WorkspaceContainer attaches a ref to its wrapped component. Without `forwardRef` here, React throws a warning: "Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?"

https://github.com/DataBiosphere/terra-ui/blob/448e871bde8e429ed1092b27969b5ee97d8df34c/src/pages/workspaces/workspace/WorkspaceContainer.js#L417